_blog/scaling-dot-net-core-microservices/index.md

```diff
@@ -0,0 +1,403 @@
+
+Using `tmux`, we can see 6 windows running each of the services we mentioned.
+All dotnet core services were started with `dotnet run` within the project directory.
+What happens if one of the services stops?  For example, let say that the worker service stops.
+The website would be able to submit a ticket to the API, and the API would put the ticket into the database.
+The API would then send the ticket to the queue, but nothing would be subscribed to pick up that ticket and process it.
+Also, we have a lot to do to get the service up and running locally, are we expecting to do this to debug the application locally?
+Is there a better way that we could manage this?
+
+## The Case For Containers
+
+So far, we've found out that running our microservice application has been a bit harder to do than we'd like.
+If we compare our services to spinning plates, we have a lot of spinning plates that are dependent on each other and we're not great at noticing when a plate has fallen.
+We're going to add a `Dockerfile` to each of our services to containerise them.
+
+### Containers should do just one thing, and one thing well
+
+Containers can be whatever you want them to be, a 30mb console application to a 4gb monolith.
+But they should really just do one job and do one job well.
+When you scale your application, do you want to scale your whole application or just some services within it?
+If a service fails or has an error, do you want to know that it was the one thing that did it, or possibly one or many of several?
+
+### Do I need to do anything differently if the code is in a container
+
+If you can build and run your code locally, you'll be able to run it within a container.
+If you're running an application and you have to manually kill threads locally, will this work within a container?
+Yes, it will!  But you're going to find problems down the line.
+
+The containers run microservices, if we implement bad code and practices, these will just grow when scaling the application.
+As with most concerns, by addressing the common issues early vast improvements can be attained later.
+
+> TOP TIP: When you build something, learn how to kill it and make it graceful
+
+Regardless of what part of your system this relates to, you should be able to create and delete parts of your system.
+For example, if you're creating a cluster, you should be able to have a command that can delete your cluster.
+If you create a console application, you should manage the `Ctrl + C` or `SIGTERM` event.
+Normally, containers will give containers 30 seconds notice before terminating an application.
+If you have a console application that is processing a lead, how do you manage this when the orchestrator fires the `Ctrl + C` event?
+Do you delete the process or wait for it to complete?  These are the decisions you should be making prior to moving to a container.
+Also, this will empower your team later on, as you'll know how to manage when you have an actual fire.
```

Comment:
This section should be refactored with the previous two. You don't explain how to run containers, but suddenly I know how to stop them. There's some knowledge missing in between.